Source/Iterators/Common/Identity.php

```diff
@@ -45,7 +45,9 @@ public static function hash($value)
 
             case 'o': //object
 
-                return 'o' . spl_object_hash($value);
+                return $value instanceof \DateTimeInterface || $value instanceof \DateTime
+                        ? 'dt:' . get_class($value) . ':' . $value->getTimestamp()
```

Comment:
@Ocramius The `identity::hash` function returns a unique string representing if two values are identical (`===`), to avoid collisions each type of value has a prefix, `dt` is just an arbitrary prefix string I choose for date times. This is used throughout the implementation of the API to compare values.